libsolidity/analysis/TypeChecker.cpp

```diff
@@ -937,6 +937,11 @@ bool TypeChecker::visit(Assignment const& _assignment)
 	_assignment.annotation().type = t;
 	if (TupleType const* tupleType = dynamic_cast<TupleType const*>(t.get()))
 	{
+		if (_assignment.assignmentOperator() != Token::Assign)
```

Comment:
No, it should be correct. please take a look at https://github.com/ethereum/solidity/pull/1749/files/77fd76cc2c06b33625d2dc1e6b602addf1033a08#diff-9f2a45efaa7b75ab7ec620d3177cf01aR954